### ISSUE TITLE
NaN's in gradient when using flex_pool and flex_deconv

### ISSUE THREAD

**Issue by wassname** [cgtuebingen/Flex-Convolution#1]
Thanks for releasing your code!

I realise you not finished releasing yet, but I had a try anyway and noticed an issue. I'm not asking you to fix it, just reporting it in case you were unaware.

When I tried a [full model](https://github.com/wassname/Flex-Convolution) I noticed that I get NaN's if I use more than one layer of pooling or deconv. It sometimes happens only after a few batches so it seems that they might include operations that introduce unstable gradients. I haven't looked at your *.cc files for the culprit though.

**Comment by PatWie** [comment]
To leave a response here: We have some unrelated issues [running your code](https://github.com/wassname/Flex-Convolution/blob/master/mnist_clouds.ipynb), which is necessary to track down where your NaN's are coming from.

Btw: We highly suggest to use `tf.keras` for [some reasons](https://stackoverflow.com/questions/51167542/tensorflow-creating-new-variables-despite-reuse-set-to-true/51167774#51167774).

**Comment by wassname** [comment]
Thanks for the example :)
